dts: riscv: Add sifive,plic-1.0.0 binding and fix riscv,ndev values

ulfalizer opened the pull request:
Add a new sifive,plic-1.0.0 binding that inherits from the riscv,plic0
binding. The new binding adds a required riscv,ndev property, which
gives the number of external interrupts supported.

Use the new binding for microsemi-miv.dtsi (with a value of 31 for
riscv,ndev, from http://www.actel.com/ipdocs/MiV_RV32IMAF_L1_AHB_HB.pdf)
and riscv32-fe310.dtsi (which already assigns riscv,ndev).

Also remove a spurious riscv,ndev assignment from
riscv32-litex-vexriscv.dtsi.

Also make edtlib and the old scripts/dts/ scripts replace '.' in
compatible strings with '_' when generating identifiers.

galak reviewed (changes requested):
I think `riscv,ndev` should only be on the sifive,plic.  It should probably be removed from `dts/riscv32/riscv32-litex-vexriscv.dtsi`

nategraff-sifive reviewed (approved):
Approving the PLIC property; I can't comment on the VexRiscv interrupt controller.

ulfalizer commented:
@galak 
Changed

galak reviewed (changes requested):
I think I was going to say we should match the Linux bindings a introduce a `sifive,plic-1.0.0` binding that has `riscv,ndev` and inherits from `riscv,plic0.yaml`

ulfalizer committed:
dts: riscv: Add sifive,plic-1.0.0 binding and fix riscv,ndev values

Add a new sifive,plic-1.0.0 binding that inherits from the riscv,plic0
binding. The new binding adds a required riscv,ndev property, which
gives the number of external interrupts supported.

Use the new binding for microsemi-miv.dtsi (with a value of 31 for
riscv,ndev, from http://www.actel.com/ipdocs/MiV_RV32IMAF_L1_AHB_HB.pdf)
and riscv32-fe310.dtsi (which already assigns riscv,ndev).

Also remove a spurious riscv,ndev assignment from
riscv32-litex-vexriscv.dtsi.

Also make edtlib and the old scripts/dts/ scripts replace '.' in
compatible strings with '_' when generating identifiers.

Signed-off-by: Ulf Magnusson <Ulf.Magnusson@nordicsemi.no>

ulfalizer commented:
@galak 
Updated

ulfalizer commented:
Got help from @PiotrZierhoffer and @mateusz-holenko re. the MiV board.

galak commented on dts/riscv32/microsemi-miv.dtsi:
```diff
@@ -44,14 +44,15 @@
 		plic: interrupt-controller@40000000 {
 			#interrupt-cells = <1>;
-			compatible = "riscv,plic0";
+			compatible = "sifive,plic-1.0.0";
```
Just checking, its the case that microsemi-miv is using SiFive PLIC?

PiotrZierhoffer replied:
Yes, that's correct

galak reviewed (commented):
You made need to update `soc/riscv32/riscv-privilege/miv/dts_fixup.h` and `drivers/interrupt_controller/plic.c`

ulfalizer commented:
> You made need to update `soc/riscv32/riscv-privilege/miv/dts_fixup.h` and `drivers/interrupt_controller/plic.c`

Done

galak commented:
How'd you get 31 for riscv,ndev on microsemi-miv.dtsi?

mateusz-holenko commented:
According to this doc: https://www.microsemi.com/document-portal/doc_download/1243781-mi-v-rv32imaf-l1-axi-handbook, page 10, Table 2 "MiV_RV32IMAF_L1_AHB Architecture" there are 31 external interrupts for MiV.

galak commented:
> According to this doc: [microsemi.com/document-portal/doc_download/1243781-mi-v-rv32imaf-l1-axi-handbook](https://www.microsemi.com/document-portal/doc_download/1243781-mi-v-rv32imaf-l1-axi-handbook), page 10, Table 2 "MiV_RV32IMAF_L1_AHB Architecture" there are 31 external interrupts for MiV.

So is NUM_IRQS incorrect in soc/riscv/riscv-privilege/miv/Kconfig.defconfig.series ?
